### PR TITLE
Fixes #15435: Forward reports to upper relays in relayd

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/15435

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: relay/sources/relayd/src/output/upstream.rs [hunk 1]
```diff
@@ -28,14 +28,46 @@
...
+    endpoint: &str,
+    path: PathBuf,
+) -> impl Future<Item = (), Error = Error> + '_ {
+    tokio::fs::read(path)
+        .map_err(|e| e.into())
+        .and_then(move |d| {
+            job_config
+                .clone()
+                .client
+                .clone()
```
Comment (ncharles) [inline relay/sources/relayd/src/output/upstream.rs]: why the two clone ?

Comment (amousset) [reply relay/sources/relayd/src/output/upstream.rs]: the first is useless, good catch! (came from a refactoring a previous version)

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

File: relay/sources/relayd/src/output/upstream.rs [hunk 2]
```diff
@@ -28,14 +28,45 @@
...
-        .basic_auth("rudder", Some("rudder"))
-        .send()
-        .unwrap();
+pub fn send_report(
+    job_config: Arc<JobConfig>,
+    path: PathBuf,
+) -> Box<dyn Future<Item = (), Error = Error> + Send> {
+    let report_span = span!(Level::TRACE, "upstream");
+    let _report_enter = report_span.enter();
+    Box::new(forward_file(job_config, "report", path))
```
Comment (ncharles) [inline relay/sources/relayd/src/output/upstream.rs]: i *think* it should be reports there

**Pull request merged**